torch/utils/cpp_extension.py

```diff
@@ -280,20 +304,78 @@ def unix_wrap_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
                     self.compiler.set_executable('compiler_so', nvcc)
                     if isinstance(cflags, dict):
                         cflags = cflags['nvcc']
-                    cflags = COMMON_NVCC_FLAGS + ['--compiler-options',
-                                                  "'-fPIC'"] + cflags + _get_cuda_arch_flags(cflags)
+                    cflags = unix_cuda_flags(cflags)
                 elif isinstance(cflags, dict):
                     cflags = cflags['cxx']
-                # NVCC does not allow multiple -std to be passed, so we avoid
-                # overriding the option if the user explicitly passed it.
-                if not any(flag.startswith('-std=') for flag in cflags):
-                    cflags.append('-std=c++14')
+                append_std14_if_no_std_present(cflags)
 
                 original_compile(obj, src, ext, cc_args, cflags, pp_opts)
             finally:
                 # Put the original compiler back in place.
                 self.compiler.set_executable('compiler_so', original_compiler)
 
+        def unix_wrap_ninja_compile(sources,
+                                    output_dir=None,
+                                    macros=None,
+                                    include_dirs=None,
+                                    debug=0,
+                                    extra_preargs=None,
+                                    extra_postargs=None,
+                                    depends=None):
+            """Compiles sources by outputting a ninja file and running it."""
+            # NB: We're using private functions on self.compiler (which is
+            # an instance of distutils.UnixCCompiler). This can be fragile,
+            # but a lot of other repos also do this
+            # (see https://github.com/search?q=_setup_compile&type=Code)
+            # so it is probably OK; we'll also get CI signal if/when
+            # we update our python version (which is when distutils can be
+            # upgraded)
```

Comment:
When you showed me this patch, you showed me how it was a copy paste from some existing code in distutils. A link to that code will be helpful for readers!

Reply:
Yeah, that's a good idea